src/core/expr/cumsum.cc

```diff
@@ -0,0 +1,120 @@
+#include "expr/fexpr_func.h"
+#include "column/virtual.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "stype.h"
+#include "python/xargs.h"
+#include "documentation.h"
+#include <numeric>
+#include <vector>
+
+namespace dt {
+namespace expr {
+
+template <typename T>
+class Column_cumsum : public Virtual_ColumnImpl{
+    private:
+      Column acol_;
+
+    public:
+      Column_cumsum(Column&& a)
+        : Virtual_ColumnImpl(a.nrows(), a.stype()),
+          acol_(std::move(a))
+
+    {
+      xassert(acol_.can_be_read_as<T>());
+    }
+
+      ColumnImpl* clone() const override {
+          return new Column_cumsum(Column(acol_));
+      }
+
+      size_t n_children() const noexcept override {
+      return 1;
+    }
+
+    const Column& child(size_t) const override {
+      return acol_;
+    }
+
+
+    bool get_element(size_t i, T* out) const override {
+      T a;
+      bool isvalid = acol_.get_element(i, &a);
+      if (isvalid) {
+        *out = static_cast<T>(acol_.nrows()); // testing a dummy output here to return no of rows
```

Comment:
I'm not sure how you say this line could return the column itself. `.nrows()` obviously cannot return a column, because it returns `size_t`. Why do you think this method returns a column?